app/containers/TextInput/FormTextInput.tsx

```diff
@@ -94,7 +95,15 @@ export default class FormTextInput extends React.PureComponent<IRCTextInputProps
 	}
 
 	get iconRight() {
-		const { iconRight, onIconRightPress, theme } = this.props;
+		const { iconRight, onIconRightPress, theme, onCancelSearch, value } = this.props;
+		if (onCancelSearch && value && value?.length > 0) {
+			return (
+				<Touchable onPress={onCancelSearch} style={[styles.iconContainer, styles.iconRight]} testID='searchbox-clear'>
+					<CustomIcon name='input-clear' size={18} color={themes[theme].auxiliaryTintColor} />
+				</Touchable>
+			);
+		}
+
 		return iconRight ? (
 			<Touchable onPress={onIconRightPress} style={[styles.iconContainer, styles.iconRight]}>
```

Comment:
Can you remove touchable from here?
There's no reason to have a button for `iconRight` and `onIconRightPress` is not used anywhere